### PR TITLE
OEL-1134: Contact form Honeypot submodule.

### PR DESCRIPTION
**Pull request by Maxfire**
### Description
Honeypot submodule for antispam added.
Tests updated accordingly.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -84,7 +86,8 @@
             "composer/installers": true,
             "cweagans/composer-patches": true,
             "drupal/core-composer-scaffold": true,
-            "phpro/grumphp": true
+            "phpro/grumphp": true,
+            "dealerdirect/phpcodesniffer-composer-installer": true
```
Comment (drishu) [inline composer.json:90]: remove this, we already have sniffs installed

File: tests/src/FunctionalJavascript/MessageFormTest.php [hunk 2]
```diff
@@ -54,6 +55,9 @@ protected function setUp(): void {
       'access site-wide contact form',
       'access corporate contact form',
     ]);
+
+    // Disable the time limit.
```
Comment (drishu) [inline tests/src/FunctionalJavascript/MessageFormTest.php:59]: pls include also an explanation in the comment as to why we need this

**Review by drishu — CHANGES_REQUESTED** [review]
Also enabled oe_contact_forms_honeypot in the runner

**Review by drishu — DISMISSED** [review]

File: modules/oe_contact_forms_honeypot/oe_contact_forms_honeypot.module [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Implements hook_form_alter().
+ *
+ * Add Honeypot features to all contact message forms.
+ */
+function oe_contact_forms_honeypot_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $build_info = $form_state->getBuildInfo();
+  if (!isset($build_info['base_form_id']) || $build_info['base_form_id'] !== 'contact_message_form') {
+    return;
+  }
```
Comment (brummbar) [inline modules/oe_contact_forms_honeypot/oe_contact_forms_honeypot.module:21]: We can use `hook_form_BASE_FORM_ID_alter()`.

File: runner.yml.dist [hunk 4]
```diff
@@ -24,6 +24,7 @@ drupal:
     # Enable the modules.
     - "./vendor/bin/drush en config_devel -y"
     - "./vendor/bin/drush en oe_contact_forms -y"
+    - "./vendor/bin/drush en oe_contact_forms_honeypot -y"
```
Comment (brummbar) [inline runner.yml.dist:27]: This we don't need to add it, let's keep it disabled by default.

File: tests/src/FunctionalJavascript/MessageFormTest.php [hunk 5]
```diff
@@ -41,6 +41,7 @@ class MessageFormTest extends WebDriverTestBase {
     'contact',
     'contact_storage',
     'oe_contact_forms',
+    'oe_contact_forms_honeypot',
```
Comment (brummbar) [inline tests/src/FunctionalJavascript/MessageFormTest.php:44]: This test is for the base module functionality. We need to write another test to check that honeypot is added correctly.

**Review by brummbar — CHANGES_REQUESTED** [review]

**Review by brummbar — APPROVED** [review]

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**